File: src/LeetCode 75/plan1/704. Binary Search.go
func search(nums []int, target int) int {
    var startIndex int = 0;
    var endIndex int = len(nums) -1;
    for startIndex < endIndex {
        var minIndex int = (startIndex+endIndex)/2;
        if(nums[minIndex] == target){
            return minIndex;
        }
        if(nums[minIndex] < target){
            startIndex = minIndex+1;
        }else{
            endIndex = minIndex ;
        }
    }
    if(nums[startIndex] == target){
        return startIndex;
    }
    return -1;
    
}

File: src/Algorithm/java/704. Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        int start = 0;
        int end = nums.length-1;
        while(start<end){
            int mid = (start+end)/2;
            if(nums[mid] == target){
                return mid;
            }
            start = nums[mid]<target?mid+1:start;
            end = nums[mid]>target?mid:end;
        }
        return nums[start]==target?start:-1;
    }
}
